sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -27,9 +31,11 @@ def __init__(self):
         self._dict = {}
         self._trim_prefixes = []
         self._client = None
+        self._clients = []
 
     @classmethod
-    def load(cls, connection_string=None, endpoint=None, credential=None, **kwargs):
+    def load(cls, connection_string=None, endpoint=None, endpoints=None, credential=None,
```

Comment:
This is breaking change. By design?

Reply:
Is it breaking as it has a default value, and is a new configuration? Also, yes. Also at some point connection-strings will be added as an option. This could be done through kwargs, but as a primary connection method it seemed correct to be here.

Reply:
Code:
AzureAppConfigProvider.load("conn_str", "end_point","credential")

old behavior:
connection_string = "conn_str"
endpoint = "end_point"
credential = "credential"

New behavior:
connection_string = "conn_str"
endpoint = "end_point"
endpoints = "credential"
credential = None

Reply:
Btw, given you are still in beta, it is totally fine to have breaking change. 

Just wanted to make sure you are aware.

Reply:
I always do:

`AzureAppConfigProvider.load(connection_str="")`

Didn't think of that. As endpoint and connection string are never set at the same time.

Reply:
Also, endpoint and endpoints are never set at the same time.

Reply:
Maybe we can make them key word only arguments?

Reply:
I guess the question would be what wouldn't be a keyword argument at that point?

Reply:
Given all of them are optional, maybe we can use 

def load(cls, *, connection_string=None, endpoint=None, endpoints=None, credential=None, **kwargs)

to make all of them keyword only.

Reply:
And if you want, you can even use function overloading to make it clearer to use. e.g. https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/core/azure-core/azure/core/pipeline/transport/_requests_basic.py#L270

Reply:
The tricky thing with overloading is `connection_string` and `endpoint` are both Strings, and `connection_strings` and `endpoints` are both lists of Strings. And only one is ever set. We would have to do some weird validation of endpoint vs connection_string which I don't like.

Reply:
If we make them keyword only, we can get rid of the ambiguity, right?

Reply:
Well then it would just be:

`def load(cls, **kwargs):`

so no different than:

`def load(cls, *, connection_string=None, endpoint=None, endpoints=None, credential=None, **kwargs):`

Reply:
Sorry, not quite clear about "no different than"?

Reply:
What I mean is that from a users perspective there is no difference between the two interfaces for a user.

Reply:
If they use IDE (pycharm, vscode, etc.), def load(cls, *, connection_string=None, endpoint=None, endpoints=None, credential=None, **kwargs): shows intellisense.

Reply:
I have changed it to be the second one, so it will have the intellisense.